EIPS/eip-7002.md

```diff
@@ -0,0 +1,372 @@
+---
+eip: 7002
+title: Execution layer triggerable exits
+description: Allows validators to trigger exits via their execution layer (0x01) withdrawal credentials
+author: Danny Ryan (@djrtwo), Mikhail Kalinin (@mkalinin), Ansgar Dietrichs (@adietrichs)
+discussions-to: https://ethereum-magicians.org/t/eip-7002-execution-layer-triggerable-exits/14195
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-05-09
+---
+
+## Abstract
+
+Adds a new *stateful* precompile that allows validators to trigger exits to the beacon chain from their execution layer (0x01) withdrawal credentials.
+
+These new execution layer exit messages are appended to the execution layer block to reading by the consensus layer.
+
+## Motivation
+
+Validators have two keys -- an active key and a withdrawal credential. The active key takes the form of a BLS key, whereas the withdrawal credential can either be a BLS key (0x00) or a execution layer address (0x01). The active key is "hot", actively signing and performing validator duties, whereas the withdrawal credential can remain "cold", only performing limited operations in relation to withdrawing and ownership of the staked ETH. Due to this security relationship, the withdrawal credential ultimately is the key that owns the staked ETH and any rewards. 
+
+As currently specified, only the active key can initiate a validator exit. This means that in any non-standard custody relationships (i.e. active key is separate entity from withdrawal credentials), that the ultimate owner of the funds -- the possessor of the withdrawal credentials -- cannot independently choose to exit and begin the withdrawal process. This leads to either trust issues (e.g. ETH can be "held hostage" by the active key owner) or insufficient work-arounds such as pre-signed exits. Additionally, in the event that active keys are lost, a user should still be able to recover their funds by using their cold withdrawal credentials.
+
+To ensure that the withdrawal credentials (owned by both EOAs and smart contracts) can trustlessly control the destiny of the staked ETH, this specification enables exits triggerable by 0x01 withdrawal credentials.
+
+Note, 0x00 withdrawal credentials can be changed into 0x01 withdrawal credentials with a one-time signed message. Thus any functionality enabled for 0x01 credentials is defacto enabled for 0x00 credentials.
```

Comment:
Consider adding a note to the motivation that withdraw credentials can have complex recovery mechanisms that simultaneously protect users from both loss and theft which are not possible to private keys (e.g., time based recovery, social recovery, survivorship, etc.).

Also worth noting that by making the validator keys unnecessary for exiting, a validator can favor "protection from theft" exclusively with their validator keys without having to concern themselves with "protection from loss".  At the moment, validators **MUST** (game theoretically) prioritize protection from loss over protection from theft and these two are often in direct opposition to each other.  This means someone could have their validator keys on disk only on their heavily locked down validator node, and if the node is destroyed they can exit to recover.  Right now this would be an absolutely terrible idea.